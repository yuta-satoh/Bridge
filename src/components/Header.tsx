import Image from "next/image";
import Link from "next/link";

export default function Header() {
    return (
        <header className="h-28 w-screen bg-orange-900">
            <div className="flex items-center justify-center h-20 w-screen bg-white absolute top-8">
                <Link href={'/'}>
                    <div>
                        <Image src="/images/logo/ロゴ候補3.png" alt="bridge-logo" width={200} height={80}/>
                    </div>
                </Link>
                <ul className="flex gap-10 mr-60 ml-30">
                    <li>商品</li>
                    <li>お知らせ</li>
                    <li>ヘルプ</li>
                </ul>
                <div className="flex gap-10">
                    <form>
                        <input className="h-8 border border-neutral-500 rounded-l pl-2.5" type="text" placeholder="何をお探しですか？" />
                        <button className="h-8 text-white bg-neutral-900 border border-neutral-900 rounded-r px-1" type="submit">検索</button>
                    </form>
                    <Link href={'/login'}>
                        <div>
                            <span className="inline-block mt-1">ログイン</span>
                        </div>
                    </Link>
                    <Link href={'/cart'}>
                        <div>
                            <span className="inline-block mt-1">カート</span>
                        </div>
                    </Link>
                </div>
            </div>
        </header>
    )
}
