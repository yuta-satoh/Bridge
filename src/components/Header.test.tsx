import renderer from 'react-test-renderer';
import Header from './Header';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

jest.mock('next/router', () => ({
    useRouter() {
      return {
        push: jest.fn(),
      };
    },
  }));

test('未ログイン', () => {
  const tree = renderer.create(<Header auth={false} />);
  expect(tree).toMatchSnapshot();
});

test('ログイン時', () => {
    const tree = renderer.create(<Header auth={true} />);
    expect(tree).toMatchSnapshot();
  });
