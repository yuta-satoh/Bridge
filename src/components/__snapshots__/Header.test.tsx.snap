// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ログイン時 1`] = `
<header
  className="body h-28 w-full"
>
  <div
    className="menu flex items-center justify-center h-20 w-screen bg-white absolute top-8"
  >
    <a
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      <div>
        <img
          alt="bridge-logo"
          className="titleiconimg"
          data-nimg="1"
          decoding="async"
          height={80}
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          src="/_next/image?url=%2Fimages%2Flogo%2Fheader_logo.png&w=384&q=75"
          srcSet="/_next/image?url=%2Fimages%2Flogo%2Fheader_logo.png&w=256&q=75 1x, /_next/image?url=%2Fimages%2Flogo%2Fheader_logo.png&w=384&q=75 2x"
          style={
            {
              "color": "transparent",
            }
          }
          width={150}
        />
      </div>
    </a>
    <ul
      className="left flex gap-10 mr-60 ml-30 whitespace-nowrap"
    >
      <li
        className="text-base"
      >
        <a
          href="/items/itemlist"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          商品
        </a>
      </li>
      <li>
        <a
          className="toRemind"
          href="/#remind"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          お知らせ
        </a>
      </li>
      <li>
        ヘルプ
      </li>
    </ul>
    <div
      className="right flex gap-10"
    >
      <form
        className="form"
        onSubmit={[Function]}
      >
        <div>
          <input
            className="h-6 sp:h-8 w-32 sp:w-48 text-xs border border-neutral-500 rounded-l pl-2"
            onChange={[Function]}
            placeholder="何をお探しですか？"
            type="text"
            value=""
          />
        </div>
        <a
          href="/items/itemlist/search?genre=北欧風&genre=ナチュラル&genre=和モダン&genre=フェミニン&category=椅子&category=テーブル&category=カーテン&category=照明&category=カーペット%2Fラグ&category=ソファ&category=収納棚&category=ベッド%2F寝具&category=小物%2F雑貨&input=&order=id.desc&page=0"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          <button
            className="searchs h-6 sp:h-8 text-white bg-neutral-900 border border-neutral-900 rounded-r px-1"
            type="submit"
          >
            <span
              className="search"
            >
              検索
            </span>
          </button>
        </a>
      </form>
      <a
        className="iconModule"
        href="/mypage"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        <img
          alt=""
          className="iconimg"
          data-nimg="1"
          decoding="async"
          height={25}
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          src="/_next/image?url=%2Fimages%2Ficon%2Flogin.png&w=64&q=75"
          srcSet="/_next/image?url=%2Fimages%2Ficon%2Flogin.png&w=32&q=75 1x, /_next/image?url=%2Fimages%2Ficon%2Flogin.png&w=64&q=75 2x"
          style={
            {
              "color": "transparent",
            }
          }
          width={25}
        />
        <div>
          <span
            className="menuLabel inline-block pt-1"
          >
            マイページ
          </span>
        </div>
      </a>
      <p>
        エラー
      </p>
    </div>
  </div>
</header>
`;

exports[`未ログイン 1`] = `
<header
  className="body h-28 w-full"
>
  <div
    className="menu flex items-center justify-center h-20 w-screen bg-white absolute top-8"
  >
    <a
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      <div>
        <img
          alt="bridge-logo"
          className="titleiconimg"
          data-nimg="1"
          decoding="async"
          height={80}
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          src="/_next/image?url=%2Fimages%2Flogo%2Fheader_logo.png&w=384&q=75"
          srcSet="/_next/image?url=%2Fimages%2Flogo%2Fheader_logo.png&w=256&q=75 1x, /_next/image?url=%2Fimages%2Flogo%2Fheader_logo.png&w=384&q=75 2x"
          style={
            {
              "color": "transparent",
            }
          }
          width={150}
        />
      </div>
    </a>
    <ul
      className="left flex gap-10 mr-60 ml-30 whitespace-nowrap"
    >
      <li
        className="text-base"
      >
        <a
          href="/items/itemlist"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          商品
        </a>
      </li>
      <li>
        <a
          className="toRemind"
          href="/#remind"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          お知らせ
        </a>
      </li>
    </ul>
    <div
      className="right flex gap-10"
    >
      <form
        className="form"
        onSubmit={[Function]}
      >
        <div>
          <input
            className="h-6 sp:h-8 w-32 sp:w-48 text-xs border border-neutral-500 rounded-l pl-2"
            onChange={[Function]}
            placeholder="何をお探しですか？"
            type="text"
            value=""
          />
        </div>
        <a
          href="/items/itemlist/search?genre=北欧風&genre=ナチュラル&genre=和モダン&genre=フェミニン&category=椅子&category=テーブル&category=カーテン&category=照明&category=カーペット%2Fラグ&category=ソファ&category=収納棚&category=ベッド%2F寝具&category=小物%2F雑貨&input=&order=id.desc&page=0"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          <button
            className="searchs h-6 sp:h-8 text-white bg-neutral-900 border border-neutral-900 rounded-r px-1"
            type="submit"
          >
            <span
              className="search"
            >
              検索
            </span>
          </button>
        </a>
      </form>
      <a
        className="iconModule"
        href="/login"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        <img
          alt=""
          className="iconimg"
          data-nimg="1"
          decoding="async"
          height={25}
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          src="/_next/image?url=%2Fimages%2Ficon%2Flogin.png&w=64&q=75"
          srcSet="/_next/image?url=%2Fimages%2Ficon%2Flogin.png&w=32&q=75 1x, /_next/image?url=%2Fimages%2Ficon%2Flogin.png&w=64&q=75 2x"
          style={
            {
              "color": "transparent",
            }
          }
          width={25}
        />
        <div>
          <span
            className="menuLabel inline-block whitespace-nowrap"
          >
            ログイン
          </span>
        </div>
      </a>
      <a
        className="iconModule"
        href="/cart"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        <img
          alt=""
          className="iconimg"
          data-nimg="1"
          decoding="async"
          height={25}
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          src="/_next/image?url=%2Fimages%2Ficon%2Fcart.png&w=64&q=75"
          srcSet="/_next/image?url=%2Fimages%2Ficon%2Fcart.png&w=32&q=75 1x, /_next/image?url=%2Fimages%2Ficon%2Fcart.png&w=64&q=75 2x"
          style={
            {
              "color": "transparent",
            }
          }
          width={25}
        />
        <span
          className="menuLabel inline-block whitespace-nowrap"
        >
          カート
        </span>
      </a>
    </div>
  </div>
</header>
`;
